expat/lib/xmlparse.c

```diff
@@ -609,14 +609,22 @@ struct XML_ParserStruct {
      macro works. */
   void *m_userData;
   void *m_handlerArg;
-  char *m_buffer;
+
+  // How the four parse buffer pointers below relate in time and space:
+  //
+  //   m_buffer <= m_bufferPtr <= m_bufferEnd  <= m_bufferLim
+  //   |           |              |               |
+  //   <--parsed-->|              |               |
+  //               <---parsing--->|               |
+  //                              <--unoccupied-->|
+  //   <---------total-malloced/realloced-------->|
```

Comment:
This is _very_ useful, and I wish it had been here when I started looking into expat. :)

Well visualized!

Reply:
@Snild-Sony cool, thanks for the review! :pray: